### ISSUE TITLE
Remove usage of `eval` to allow safer Content Security Policy (CSP)

### ISSUE THREAD

**Issue by tsvetomir** [telerik/kendo-angular#1005]
## I'm submitting a...
* Suggestion for improvement

## Current behavior
The Data Query and Grid packages make use of `new Function` (eval) to build accessors for data item fields. As a result, a [Content Security Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) must include `unsafe-eval`. This severely degrades the practical security benefits of the applied policy.

## Expected behavior
The components should not require `unsafe-eval`.

## Minimal reproduction of the problem with instructions
1. Clone the [Kendo Angular Quickstart-CLI](https://github.com/telerik/kendo-angular-quickstart-cli) repo

    `git clone https://github.com/telerik/kendo-angular-quickstart-cli.git`
1. Checkout the [test-csp branch](https://github.com/telerik/kendo-angular-quickstart-cli/compare/test-csp)

    `git checkout test-csp`
1. Build the application with AoT

    `ng serve --aot true`
1. Load the page at http://localhost:4200

![image](https://user-images.githubusercontent.com/247917/31534794-1b6984a2-b001-11e7-8691-c50d5a39e673.png)

## What is the motivation or use case for changing the behavior?
Improve the security of applications using Kendo UI for Angular

## Environment

Package versions:

```
kendo-angular2-quickstart-cli@0.0.0
├── @angular/animations@4.4.5
├── @angular/cli@1.4.7
├── @angular/common@4.4.5
├── @angular/compiler@4.4.5
├── @angular/compiler-cli@4.4.5
├── @angular/core@4.4.5
├── @angular/forms@4.4.5
├── @angular/http@4.4.5
├── @angular/platform-browser@4.4.5
├── @angular/platform-browser-dynamic@4.4.5
├── @angular/router@4.4.5
├── @progress/kendo-angular-buttons@1.1.0
├── @progress/kendo-angular-dateinputs@1.2.1
├── @progress/kendo-angular-dropdowns@1.2.2
├── @progress/kendo-angular-excel-export@1.0.4
├── @progress/kendo-angular-grid@1.5.0
├── @progress/kendo-angular-inputs@1.3.1
├── @progress/kendo-angular-intl@1.2.2
├── @progress/kendo-angular-l10n@1.0.4
├── @progress/kendo-data-query@1.0.7
├── @progress/kendo-drawing@1.4.0
├── @progress/kendo-theme-default@2.43.1
├── @types/jasmine@2.5.38
├── @types/node@6.0.89
├── codelyzer@2.0.1
├── core-js@2.5.1
├── jasmine-core@2.5.2
├── jasmine-spec-reporter@3.2.0
├── karma@1.4.1
├── karma-chrome-launcher@2.0.0
├── karma-cli@1.0.1
├── karma-coverage-istanbul-reporter@0.2.3
├── karma-jasmine@1.1.0
├── karma-jasmine-html-reporter@0.2.2
├── protractor@5.1.2
├── rxjs@5.4.3
├── ts-node@2.0.0
├── tslint@4.5.1
├── typescript@2.2.2
└── zone.js@0.8.18
```

Browser:

- Chrome (desktop) version 61

**Comment by tsvetomir** [comment]
The latest dev version of the `@progress/kendo-data-query` package resolves the initial error. Update with:

`npm install --save @progress/kendo-data-query@dev`

We identified another usage of eval within the same package - the [filterBy method](https://www.telerik.com/kendo-angular-ui/components/dataquery/#toc-filtering), also used by [process](https://www.telerik.com/kendo-angular-ui/components/dataquery/#toc-process-helpers-for-bulk-operations). We're researching the possibility to replace it with a static version, but for the time being filtering should be carried out by other means. Not that this code is not called directly by the components, but is instead shipped as a helper.

**Comment by tsvetomir** [comment]
I've updated the test branch to the latest version of Data Query. The only issue at the moment is with filtering and can be reproduced by trying to apply a filter on any column. See notes above.

**Comment by tsvetomir** [comment]
The latest development version of `@progress/kendo-data-query` fixes all CSP errors.

**Comment by tsvetomir** [comment]
Shipped in v1.1.1
